src/ConfigSourceAwareTrait.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Dhii\Modular\Locator;
+
+/**
+ * Functionality that allows setting and retrieval of a module configuration source (protected).
+ *
+ * @since [*next-version*]
+ */
+trait ConfigSourceAwareTrait
+{
+    /**
```

Comment:
Was trying to avoid constraints to the property signature. But you are right, the stricter - the better.